monai/inferers/splitter.py

```diff
@@ -58,142 +92,369 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
+                or a filename (str) or list of filenames to the image(s).
+
+        Returns:
+            padded_spatial_shape
+
+        """
+        spatial_shape = self.get_input_shape(inputs)
+        if not self.pad_mode:
+            return spatial_shape
+        spatial_ndim = len(spatial_shape)
+        patch_size, overlap, offset = self._get_valid_shape_parameters(spatial_shape)
+        pad_size, _ = self._calculate_pad_size(spatial_shape, spatial_ndim, patch_size, offset, overlap)
+        padded_spatial_shape = tuple(ss + ps + pe for ss, ps, pe in zip(spatial_shape, pad_size[1::2], pad_size[::2]))
+
+        return padded_spatial_shape
+
+    def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
         """Split the input tensor into patches and return patches and locations.
 
         Args:
-            inputs: a torch.Tensor with BCHW[D] dimensions, representing an image or a batch of images.
+            inputs: either a torch.Tensor with BCHW[D] dimensions, representing an image or a batch of images
 
         Yields:
             tuple[torch.Tensor, Sequence[int]]: yields tuple of patch and location
         """
-        n_non_spatial_dims = 2
-        spatial_ndim = inputs.ndim - n_non_spatial_dims
-        spatial_shape = inputs.shape[n_non_spatial_dims:]
-        patch_size = self._get_valid_patch_size(spatial_ndim)
-        overlap = self._get_valid_overlap(spatial_ndim, patch_size)
-        offset = self._get_valid_offset(spatial_shape, spatial_ndim, patch_size)
+
+        if not isinstance(inputs, torch.Tensor):
+            raise ValueError(f"The input should be a tensor. {type(inputs)} is given.")
+
+        spatial_shape = inputs.shape[2:]
+        spatial_ndim = len(spatial_shape)
+        patch_size, overlap, offset = self._get_valid_shape_parameters(spatial_shape)
         pad_size, is_start_padded = self._calculate_pad_size(spatial_shape, spatial_ndim, patch_size, offset, overlap)
 
-        if any(pad_size):
+        # Padding
+        if self.pad_mode and any(pad_size):
             # pad the inputs
-            inputs = torch.nn.functional.pad(
-                inputs, pad_size[::-1], look_up_option(self.pad_mode, PytorchPadMode).value, **self.pad_kwargs
-            )
+            inputs = torch.nn.functional.pad(inputs, pad_size[::-1], mode=self.pad_mode, value=self.pad_value)
             # update spatial shape
-            spatial_shape = inputs.shape[n_non_spatial_dims:]
+            spatial_shape = inputs.shape[2:]
+            # correct the offset with respect to the padded image
+            if is_start_padded:
+                offset = tuple(off + p for off, p in zip(offset, pad_size[1::2]))
+
+        # Splitting
+        for location in iter_patch_position(spatial_shape, patch_size, offset, overlap, False):
+            patch = self._get_patch(inputs, location, patch_size)  # type: ignore
+            patch = ToTensor(device=self.device)(patch)  # type: ignore
```

Comment:
I'd suggest calling `convert_to_tensor` directly here rather than using a transform. When you do need a transform I'd also recommend not instantiating it just to call but keep an instance around and reuse it to avoid creating objects.